src/main/java/com/allocadia/carbonite/CarbonitePersist.java

```diff
@@ -0,0 +1,111 @@
+        persistDirty(modifiedObjects.getDirtyObjects(), info);
+    }
+
+    @SneakyThrows
+    private void persistDirty(List<PersistedObject> dirtyObjects, PersistenceInfo<PersistedObject> info) {
+        for (int i = 0; i < dirtyObjects.size(); i++) {
+            PersistedObject dirtyObject = dirtyObjects.get(i);
+            List<String> dirtyFields = new ArrayList<>(dirtyObject.getDirty());
+            
+            String updateSql = Joiner.on("\n").join(
+                "UPDATE " + info.getTableName(),
+                "SET " + getFieldUpdateStmt(dirtyFields, info),
+                "WHERE " + info.getField2Column().get(info.getIdField()) + " = ?");
+            
+            Statement statement = super.getJdbcTemplate().getDataSource().getConnection().createStatement();
+            statement.add
+        }
+    }
+
+    private String getFieldUpdateStmt(List<String> dirtyFields, PersistenceInfo<PersistedObject> info) {
+        Map<String, Field> fieldName2Field = info.getFieldName2Field();
+        Map<Field, String> field2Column = info.getField2Column();
+        
+        List<String> columnStmts = dirtyFields.stream()
+            .map(f -> {
+                Field field = fieldName2Field.get(f);
+                return field2Column.get(field);
+            })
+            .map(cn -> cn + " = ?")
+            .collect(Collectors.toList());
+        
+        return String.join(", ", columnStmts);
+    }
+
+    @SneakyThrows
+    private void persistNew(List<PersistedObject> newObjects, PersistenceInfo<PersistedObject> info) {
+        for (int i = 0; i < newObjects.size(); i++) {
+            PersistedObject newObject = newObjects.get(i);
+            List<String> dirtyFields = new ArrayList<>(newObject.getDirty());
+            String insertSql = Joiner.on("\n").join(
+                "INSERT INTO " + info.getTableName(),
+                "(" + String.join(", ", newObject.getDirty()) + ")",
+                "VALUES",
+                "(" + QueryUtils.createParamList(newObject.getDirty().size()) + ")");
+            
+            KeyHolder keyHolder = new GeneratedKeyHolder();
+            super.getJdbcTemplate().update(
+                new PreparedStatementCreator() {
+                    
+                    @SneakyThrows
+                    @Override
+                    public PreparedStatement createPreparedStatement(Connection con) throws SQLException {
+                        PreparedStatement ps = con.prepareStatement(insertSql);
+                        for (int i = 0; i < dirtyFields.size(); i++) {
+                            Field field = info.getFieldName2Field().get(dirtyFields.get(i));
+                            ps.setObject(i, field.get(newObject));
+                        }
+                        return ps;
+                    }
+                },
+                keyHolder);
+            
+            PersistedObject persistedObject = super.getJdbcTemplate().query(
```

Comment:
`.queryForObject` instead of `.query` will return a single object and throw if there is 0 or >1 results